src/models/fragmentation_baseline.py

```diff
@@ -0,0 +1,249 @@
+            should be performed. Defaults to None.
+        target_crs (Optional[Union[str, pyproj.CRS]], optional): The coordinate
+            reference system to use for the class metric computation. For interpretable
+            results a CRS with units of meter (e.g. UTM) should be used.
+            Defaults to None.
+        target_x_res (float, optional): The target pixel resolution along the
+            x-direction in the target coordinate reference system. If the CRS has units
+            of meter, this corresponds to meters per pixel. Up/downsampling is
+            performed via nearest-neighbor sampling with rasterio. Defaults to 300.
+        target_y_res (float, optional): The target pixel resolution along the
+            y-direction in the target coordinate reference system. If the CRS has units
+            of meter, this corresponds to meters per pixel. Up/downsampling is
+            performed via nearest-neighbor sampling with rasterio. Defaults to 300.
+        no_data (int, optional): The no-data value for the landcover data.
+            Defaults to 0.
+        rois_index_col (str, optional): Name of the attribute that will distinguish
+            region of interest in `rois`. Defaults to "name".
+        save_path (Union[str, os.PathLike, None]): The path at which to save the outcome
+            of the analysis.
+        **kwargs: Keyword arguments of the `compute_class_metrics_df` of pylandstats
+
+    Returns:
+        xr.DataArray: A multi-dimensional array with axes (zones, landcover classes,
+            metrics, dates) which contains all fragmentation statistics. Values for
+            classes that were not present in the data are `np.nan`
+    """
+
+    if save_path is not None:
+        save_path = pathlib.Path(save_path)
+        assert save_path.parent.exists(), "Save path directories does not exist."
+        assert not save_path.exists(), "File already exists. Override not supported."
+    assert rois is None or isinstance(rois, gpd.GeoDataFrame)
+
+    fragmentation_statistics = {}
+    for date, landcover in tqdm(landcover_data.items()):
+        logging.info("Analysing data for %s", date)
+        fragmentation_statistics[date] = _analyse_fragmentation(
+            landcover=landcover,
+            rois=rois,
+            target_crs=target_crs,
+            target_x_res=target_x_res,
+            target_y_res=target_y_res,
+            no_data=no_data,
+            rois_index_col=rois_index_col,
+        )
+
+    # Convert final data to xarray
+    logging.info("Analysis completed. Converting data to xarray.")
+    fragmentation_data = _fragmentation_dataframes_to_xarray(
+        fragmentation_statistics=fragmentation_statistics,
+        landcover_classes=landcover_classes,
+        rois=rois,
+        rois_index_col=rois_index_col,
+    )
+
+    # Save data to disk for further analysis
+    if save_path is not None:
+        fragmentation_data.to_netcdf(save_path)
+        save_path.chmod(0o664)
```

Comment:
This makes the file read-only for others, right? Maybe we should add this to the save code in `geograph.py`?

Reply:
This changes the file permissions to `-rw-rw-r--` (owner read & write, group read & write, others read)
(By default stuff on the gws is created as `-rw-r--r--` (644).

I like [this site](https://chmodcommand.com/chmod-664/) btw - super useful for crafting chmod commands 😉

Reply:
Yes, controlling permissions for geograph could be useful - do you know if ppl need to have `execute` right to load a pickle file? (then we might want `6644` or `6664`)

Reply:
I believe you don't need `execute` to load pickle files. But do you know if there is any practical difference on JASMIN for 664 vs 644 (i.e. are all users in a gws part of the same group)?

Reply:
Yes, I think all users of the `gws` are in the same group as far as I understand. I couldn't rename some 644 files for example, but could rename 664 ones.